core/primitives/src/runtime/parameter_table.rs

```diff
@@ -185,36 +182,49 @@ impl ParameterTable {
         key: Parameter,
     ) -> Result<T, InvalidConfigError> {
         let value = self.parameters.get(&key).ok_or(InvalidConfigError::MissingParameter(key))?;
-        serde_json::from_value(value.clone()).map_err(|parse_err| {
+        serde_yaml::from_value(value.clone()).map_err(|parse_err| {
             InvalidConfigError::WrongValueType(
                 parse_err,
                 key,
                 std::any::type_name::<T>(),
-                value.to_string(),
+                value.clone(),
             )
         })
     }
 
     /// Read and parse a parameter from the `ParameterTable`.
     fn get_u128(&self, key: Parameter) -> Result<u128, InvalidConfigError> {
         let value = self.parameters.get(&key).ok_or(InvalidConfigError::MissingParameter(key))?;
-
-        near_primitives_core::serialize::dec_format::deserialize(value).map_err(|parse_err| {
-            InvalidConfigError::WrongValueType(
-                parse_err,
-                key,
-                std::any::type_name::<u128>(),
-                value.to_string(),
-            )
-        })
+        match value {
+            // Values larger than u64 are stored as quoted strings, so we parse them as YAML
+            // document to leverage deserialization to u128.
+            serde_yaml::Value::String(v) => serde_yaml::from_str(v).map_err(|parse_err| {
+                InvalidConfigError::WrongValueType(
+                    parse_err,
+                    key,
+                    std::any::type_name::<u128>(),
+                    value.clone(),
+                )
+            }),
+            // If the value is a number (or any other type), the usual conversion should work.
+            _ => serde_yaml::from_value(value.clone()).map_err(|parse_err| {
+                InvalidConfigError::WrongValueType(
+                    parse_err,
+                    key,
+                    std::any::type_name::<u128>(),
+                    value.clone(),
+                )
+            }),
+        }
     }
 
-    fn fee_json(&self, key: FeeParameter) -> serde_json::Value {
-        json!( {
-            "send_sir": self.get(format!("{key}_send_sir").parse().unwrap()),
-            "send_not_sir": self.get(format!("{key}_send_not_sir").parse().unwrap()),
-            "execution": self.get(format!("{key}_execution").parse().unwrap()),
-        })
+    fn fee_yaml(&self, key: FeeParameter) -> serde_yaml::Value {
+        serde_yaml::to_value(BTreeMap::from([
+            ("send_sir", self.get(format!("{key}_send_sir").parse().unwrap())),
+            ("send_not_sir", self.get(format!("{key}_send_not_sir").parse().unwrap())),
+            ("execution", self.get(format!("{key}_execution").parse().unwrap())),
+        ]))
+        .expect("failed to construct fee yaml")
```

Comment:
we may also want to reconsider this construct, but not the highest priority on my list

Reply:
For this one, I plan to parse and store it as a structured parameter
```rust
struct Fee {
  send_sir: Gas,
  send_not_sir: Gas,
  execution: Gas,
}
```
Will working on this and weights representation in the next PR.

The main challenge here will be to keep the diff-config merging code relatively simple, but I think that should be doable.